lib/SplatArgPass.cpp

```diff
@@ -135,10 +135,11 @@ void SplatArgPass::replaceCall(Function *NewCallee, CallInst *Call) {
 }
 
 bool SplatArgPass::runOnModule(Module &M) {
-  std::list<Function *> func_list;
-  for (auto &F : M) {
+  std::list<std::pair<Function *, const Builtins::FunctionInfo &>> func_list;
+  // Collect candidates
+  for (auto &F : M.getFunctionList()) {
     // process only function declarations
-    if (F.empty() && !F.use_empty()) {
+    if (F.isDeclaration() && !F.use_empty()) {
       auto &func_info = Builtins::Lookup(&F);
       auto func_type = func_info.getType();
       switch (func_type) {
```

Comment:
The `default` clause in this switch seems to have been dropped for some reason:
```warning: 289 enumeration values not handled in switch:```

Reply:
Great, I will add the default clause back as well.